x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/index.tsx

```diff
@@ -348,7 +348,8 @@ export const EditPackagePolicyForm = memo<{
   const [formState, setFormState] = useState<PackagePolicyFormState>('INVALID');
   const savePackagePolicy = async () => {
     setFormState('LOADING');
-    const result = await sendUpdatePackagePolicy(packagePolicyId, packagePolicy);
+    const { elasticsearch, ...restPackagePolicy } = packagePolicy; // ignore 'elasticsearch' property since it fails route validation
+    const result = await sendUpdatePackagePolicy(packagePolicyId, restPackagePolicy);
```

Comment:
@elastic/fleet: this is to overcome this error we saw when trying to update settings for the integration policy:

![Screen Shot 2021-12-02 at 11 07 59 PM](https://user-images.githubusercontent.com/1967266/144679149-3fce0cd2-3c26-4d3a-94e2-bd50f15e96a9.png)

Should this be necessary, or is this a bug? I'm not even sure how `elasticsearch` became a field on the integration policy object in the first place.

Reply:
This sounds like a bug to me and we should figure out where this is coming from before making a change like this

Reply:
We added the `elasticsearch` field to the package policy to store the elasticsearch privileges from the package APM is the only package to use that, so it's probably why you have this issue. 

@joshdover  I am not sure what is the better fix for that this solution works but it's not totally future proof and make the package policy API hard to consume externally, should we remove the `elasticsearch` key when we return a package policy from an API as this is an internal we use to build the agent policy later?

Reply:
Got it, now this makes more sense. I wonder if instead we should just accept these values in our validation but ignore them in the PUT and POST APIs. That way an API consumer can easily do a GET -> edit -> PUT without having to worry about these fields, but we can still return them on the GET API for simple access. We would need to document them though or include a `warning` field in the response that those fields are ignored on updates.

Reply:
I was suggesting to totally remove them form the API response so we do not have to document why there are not documented, as I do not think there is any user value to see these computed fields.

Reply:
Is there an action item here blocking merge from the Fleet side of things?

Reply:
@kpollich No I think we can go ahead with that, and create an issue to have a better solution later.